### PR TITLE
Support for status of zone cleanup

### PR DESCRIPTION
**Pull request by pszafer**
Vacuum object has status of which zone is currently clean up.

From there it is going right now only to mqtt as I don't know where to show this value in webui.
Also I added extra webapi key url as I don't want to mess up map handling without zone name.

### CODE REVIEW

**Comment by Hypfer** [comment]
I don't really understand what this should do :(

**Comment by pszafer** [comment]
Show which zone is currently cleaned up.
For example with these we can have sensors in HA and know which zone was cleaned and when.

**Comment by pszafer** [comment]
Maybe I will give you my situation as example.

Part 1:
Without this patch i have 2 input_selects.
```
vacuum_current_state:
 - bedroom
 - bathroom
 - docked
vacuum_action:
- bedroom
- bathroom
- docked
```

and then automation which is triggered by vacuum_action change -> send mqtt command to vacuum with zone name.

second automation also trigerred by vacuum_action change, delayed by 2 seconds, checks if vacuum is cleaning and then change vacuum_current_state.
same automation as above is triggered by vacuum.rockrobo state docked and then both input_selects to docked.

Part 2:
With above I use Telegram bot to "talk" with vacuum. Telegram sending message if vacuum can clean bedroom or bathroom, if I click bedroom, then it is cleaning bedroom.
After docked bot sending message if I want clean bedroom now.


With this PR I can delete part 1 and don't have to mirror zone names in multiple places (valetudo, input_select)

**Comment by Hypfer** [comment]
Ah! But why the new `/api/start_zone_cleanup` ?

**Comment by pszafer** [comment]
If you want to start zone cleanup from map page from webui then it is not necessary to provide zone name.
I can change it to some name like `mapCleanup` then we don't need extra API URL.

**Comment by Hypfer** [comment]
Oh the homepage is currently using `/api/start_cleaning_zone` and sends the raw zone coordinates.
Thats why I was a bit confused.

**Comment by pszafer** [comment]
I changed it on homepage, so it sends zone name as well, but I didn't want to change this code on map page as I think current way is most obvious.
I could leave old name and just ignore zone name if it is not provided if you think it is better way to do it.

**Comment by pszafer** [comment]
I can change this in map.html

```
document.getElementById("start_zoned_cleanup").onclick = () => {
            const repeatNumber = 1;
            const zones = map.getLocations().zones.map(zoneCoordinates => [...zoneCoordinates, repeatNumber]);
            zoned_cleanup(zones);
        }
```

to something like 
```
const zones = map.getLocations().zones.map(zoneCoordinates => [...zoneCoordinates, "mapZone", repeatNumber]);
```

then we can have one api call for all zone cleaning function.

File: lib/miio/Vacuum.js [hunk 1]
```diff
@@ -584,6 +595,15 @@ Vacuum.prototype.goTo = function(x_coord, y_coord, callback) {
     this.sendMessage("app_goto_target", [x_coord, 51200 - y_coord], {}, callback)
 };
 
+Vacuum.prototype.zoneCleanup = function(zones, zoneNames, callback){
+    this.zoneCleaningStatus = zoneNames;
+    this.startCleaningZone(zones, callback);
```
[the remark this answers is not in the document]
Comment (pszafer) [reply lib/miio/Vacuum.js]: Do I have access to valetudo configuration in Vacuum.js?
If not do you think I should add valetudo config to Vacuum.js?

[the remark this answers is not in the document]
Comment (Hypfer) [reply lib/miio/Vacuum.js]: It's not available right now but  you can just add a reference to it in the constructor.
It does make sense to have it there as well

File: lib/miio/Vacuum.js [hunk 2]
```diff
@@ -62,6 +64,15 @@ const Vacuum = function(options) {
             self.codec.setToken(self.token);
         }
     });
+
+    this.events.on("miio.status", (statusData) => {
+        if (this.zoneCleaningStatus && statusData.battery !== undefined){
+            let state = statusData.state
+            if (state != 17 && state != 10){
```
[the remark this answers is not in the document]
Comment (pszafer) [reply lib/miio/Vacuum.js]: I can move these magic to global conts in another PR
Just tell me if `json` +
```
const GLOBALS = require("./global_vars");
```

is good approach for you?
Is it possible to load these globals then in *.html as well?

[the remark this answers is not in the document]
Comment (Hypfer) [reply lib/miio/Vacuum.js]: Yeah JSON would be fine I think. Not sure how to achieve the web part with the current UI but that should be also possible using a JSON file

[the remark this answers is not in the document]
Comment (pszafer) [reply lib/miio/Vacuum.js]: Ok, when this will be merged I'll create global_vars PR.

**Pull request merged**